sequencer/src/enriched_handle.rs

```diff
@@ -0,0 +1,55 @@
+use hotshot::{traits::NodeImplementation, types::SystemContextHandle};
+use hotshot_types::{light_client::StateSignature, traits::node_implementation::NodeType};
+use std::{
+    ops::{Deref, DerefMut},
+    sync::{Arc, RwLock},
+};
+
+use crate::api::signature_pool::StateSignaturePool;
+
+pub struct EnrichedSystemContextHandle<TYPES: NodeType, I: NodeImplementation<TYPES>> {
+    handle: SystemContextHandle<TYPES, I>,
+
+    state_signature_pool: Arc<RwLock<StateSignaturePool>>,
+}
+
+impl<TYPES: NodeType, I: NodeImplementation<TYPES>> Deref
```

Comment:
Nit: this usage of `Deref` is [frowned upon](https://rust-lang.github.io/api-guidelines/predictability.html?highlight=deref#only-smart-pointers-implement-deref-and-derefmut-c-deref). Better to use `AsRef` or just an inherent method like `fn hotshot(&self) -> &SystemContextHandle`